### PR TITLE
Example stencil project, upgrade to webpack2.

### PR DESCRIPTION
**Pull request by dzearing**
This is the start of project scaffolding. It's not 100% complete, I'd like to get this part in though. 

The idea will be that we'll have a project which can be xcopied for a number of projects. I've pushed some changed into example-app-base to make it resemble react storybook, but still use our bits and pieces in the pipeline.

Still more to check in.

### CODE REVIEW

File: packages/utilities/src/css.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ export interface ICssMapping {
   [className: string]: boolean;
 }
 
-export type ICssInput = string | ICssMapping;
+export type ICssInput = string | ICssMapping | undefined | boolean;
```
Comment (cschleiden) [inline packages/utilities/src/css.ts]: why a boolean?

File: packages/utilities/src/css.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export function css(...args: ICssInput[]) {
       if (typeof arg === 'string') {
         classes.push(arg);
       } else {
-        for (let key in arg) {
+        for (let key in arg as any) {
```
Comment (cschleiden) [inline packages/utilities/src/css.ts]: can't we cast to `ICssMapping`?

File: packages/utilities/src/css.ts (hunk 1 above)
Comment (dzearing) [inline packages/utilities/src/css.ts]:
```tsx
css(
  !!booleanResult && style.foo,
)
```

Honestly it should be null | undefined | boolean

File: packages/utilities/src/css.ts (hunk 2 above)
Comment (dzearing) [inline packages/utilities/src/css.ts]: Maybe so; I honestly struggled with the typings here. I will try.

File: packages/utilities/src/css.ts (hunk 1 above)
Comment (dzearing) [inline packages/utilities/src/css.ts]: Adding null.

File: packages/utilities/src/css.ts (hunk 2 above)
Comment (dzearing) [inline packages/utilities/src/css.ts]: Tried, got a strange typings error; the right hand side must be any.

**Pull request merged**